### PR TITLE
Create explore page base

### LINKED ISSUE #48: Add "Explore page"
Add "Explore page" - a page that displays all the questions in the app.

### PR DESCRIPTION
**Pull request by danitLevi**
- creating explore page which displays all questions in the app.
- update model.py file:
  * creating  "get_answers_num" method in Question model
- updating the url.py file and create an "explore" URL.
- updating the view.py file and creating QuestionsListView class
- creating explore.html template file
  * Clicking on question's title leads to the selected question page
  * Clicking on explore button in menu bar leads to explore page
- adding css file
- adding tests 

fix #48

### CODE REVIEW

**Comment by danitLevi** [comment]
@ifireball  @AviadP
This PR is ready for review

File: Pipfile [hunk 1]
```diff
@@ -7,10 +7,12 @@ verify_ssl = true
 flake8 = "*"
 pytest = "*"
 pytest-django = "*"
+django-filter = "*"
```
Comment (ifireball) [inline Pipfile:10]: not seeing a mention of this change in the PR comment - why is this needed?

File: home/tests.py [hunk 2]
```diff
@@ -96,3 +94,43 @@ def question_tag_test_data(question_test_data):
     new_pair.tag = test_tag
     new_pair.save()
     return test_tag
+
```
Comment (ifireball) [inline home/tests.py:97]: You'd probably want to group tests in different classes

File: home/views.py [hunk 3]
```diff
@@ -7,3 +9,10 @@ def about(request):
 
 def landingpage(request):
     return render(request, 'home/landingpage.html')
+
+
+class QuestionsListView(FilterView):
```
Comment (ifireball) [inline home/views.py:14]: Not sure how can you wrote tests to the filter view, probably by using a client that posts various filter selections and verifies the results, but you need to have tests that verify that all the filtering options you intended to provide are indeed being provided.

**Review by ifireball — CHANGES_REQUESTED** [review]

File: Pipfile (hunk 1 above)
Comment (danitLevi) [reply Pipfile:10]: I 

> not seeing a mention of this change in the PR comment - why is this needed?

@ifireball 
I mentioned it in the last commit here. 
I needed it because I used filter view for the questions view in order to enable filtering questions in the future  (I split this PR so the form that enables filtering questions is in PR  #83  ).

File: home/views.py (hunk 3 above)
Comment (danitLevi) [reply home/views.py:14]: > Not sure how can you wrote tests to the filter view, probably by using a client that posts various filter selections and verifies the results, but you need to have tests that verify that all the filtering options you intended to provide are indeed being provided.

@ifireball 
I wrote tests for the filter view in PR #83 (In this PR there is no option to filter questions yet because it contains only the base of the explore page ) .

Comment (ifireball) [reply home/views.py:14]: If the filter view isn't doing anything at this point, why introduce it to the code? Might as well make the code simpler for now and only add it when you need it and you can test it.

File: home/tests.py [hunk 4]
```diff
@@ -96,3 +94,43 @@ def question_tag_test_data(question_test_data):
...
+    assert explore_page_response.status_code == 200
+
+
+@pytest.mark.django_db
+def test_explore_page_template(explore_page_response):
+    assert explore_page_response.templates[0].name == "home/explore.html"
+
+
+@pytest.mark.django_db
+def test_explore_page_context(explore_page_response):
```
Comment (ifireball) [inline home/tests.py:131]: Its better to use client to test views (in this case via the explorer_page_response fixture you've wrote) instead of manually instantiating internal view objects. The way the test is written right now is exposing too many internal implementation details about the view, you won't be able to refactor the view code without changing the test.

File: home/tests.py (hunk 2 above)
Comment (danitLevi) [reply home/tests.py]: @ifireball
Done

File: home/views.py (hunk 3 above)
Comment (danitLevi) [reply home/views.py]: @ifireball 
Ok I convert it into ListView instead of Filter View

File: home/tests.py (hunk 4 above)
Comment (danitLevi) [reply home/tests.py]: @ifireball 
I changed it as you requested

**Comment by danitLevi** [comment]
@ifireball @AviadP
This PR is ready for re-review

File: home/static/home/explore-page.css [hunk 5]
```diff
@@ -0,0 +1,24 @@
+
```
Comment (liornoy) [inline home/static/home/explore-page.css:1]: Danit, I made a 'CSS' folder inside the static/home. I think it will be more organized if you move this file there,
sorry for not notifying you about it earlier

Comment (danitLevi) [reply home/static/home/explore-page.css]: @liornoy 
I moved it to the css folded

**Comment by danitLevi** [comment]
@ifireball @AviadP
This PR is ready for re-review

File: home/templates/home/explore.html [hunk 6]
```diff
@@ -0,0 +1,47 @@
+{% extends "home/base.html" %}
+{% load static %}
+{% block content %}
+
+<script src="//ajax.googleapis.com/ajax/libs/jquery/1.9.1/jquery.min.js"></script>
+<script type="text/javascript" src="{% static 'home/js/explore_page_js.js' %}"></script>
+<link rel="stylesheet" type="text/css" href="{% static 'home/CSS/explore-page.css' %}">
+<body class="landing-bg">
```
Comment (liornoy) [inline home/templates/home/explore.html:8]: About the class landing-bg. it's a very light color used for the landing page.
I think you should change it to the 'main-bg' class so it will match the design in the Wireframe, and have
identical bg color to the display question page

Comment (danitLevi) [reply home/templates/home/explore.html]: @liornoy 
Done

File: home/tests.py [hunk 7]
```diff
@@ -94,9 +94,9 @@ class TestAnswersManipulations:
         def answers(self):
             profile = Profile.objects.first()
             question = Question.objects.get(id=3)
-            ans1 = Answer(profile=profile, question=question, content='Answer 1', publish_date=datetime(2021, 4, 1,),
```
Comment (ifireball) [inline home/tests.py:97]: Please remove all the unnecessary whitespace changes from the PR - it makes reviewing it harder.

File: home/tests.py [hunk 8]
```diff
@@ -205,3 +205,41 @@ def tags_response(self, client):
...
+
+    class TestExplorePageView:
+
+        @pytest.mark.django_db
+        @pytest.fixture
+        def explore_page_response(self, client):
+            url = reverse('explore-page')
+            response = client.get(url)
+            return response
+
```
Comment (ifireball) [inline home/tests.py:233]: all the tests below just inspect different aspects of the same response you get from doing a plain request for the page - they can probably be together in the same test.

**Comment by danitLevi** [comment]
@ifireball @AviadP
I did the requested changes and this PR is ready for re-review

**Comment by rebeccaTubman** [comment]
LGTM

**Review by ifireball — APPROVED** [review]
